packages/email/email.js

```diff
@@ -246,11 +249,18 @@ Email.send = function(options) {
     customTransport({ packageSettings, ...options });
     return;
   }
-  if (
-    Meteor.isProduction ||
-    process.env.MAIL_URL ||
-    Meteor.settings.packages?.email
-  ) {
+
+  const mailUrlEnv = process.env.MAIL_URL;
+  const mailUrlUrlSettings = Meteor.settings.packages?.email;
+  if (Meteor.isProduction || mailUrlEnv || mailUrlUrlSettings) {
```

Comment:
I find this code a bit confusing, wouldn't be more clear if you do the validation first? Something like:

```javascript
  // This check is mostly necessary when using the flag --production when running locally.
  // And it works as a reminder to properly set the mail URL when running locally.
  if (Meteor.isProduction && !mailUrlEnv && !mailUrlUrlSettings) {
    throw new Error(
      'You do not provided a mail URL. You can provide it by using the environment variable MAIL_URL or using your settings. You can read more about it here: https://docs.meteor.com/api/email.html.'
    );
  }

  if (Meteor.isProduction || mailUrlEnv || mailUrlUrlSettings) {
    const transport = getTransport();
    smtpSend(transport, options);
    return;
   }
   devModeSend(options);
};
```